parser/src/json/schema.rs

```diff
@@ -247,6 +251,7 @@ struct Context<'a> {
     // TODO: actually use this to handle draft-specific behavior
     draft: Draft,
     defs: Rc<RefCell<HashMap<String, Schema>>>,
+    seen: Rc<RefCell<HashSet<String>>>,
```

Comment:
at some point, you will probably want `Rc<RefCell<InnerContext>>` (probably when adding normalize counters)

Reply:
Oh thanks, I'll look into it. Was just about to start adding said counters to track the number of Schema nodes that are being constructed. Probably not to normalize itself though, as I just made that "shallow".